tests/test_filters.py

```diff
@@ -361,3 +368,21 @@ def test_power_filter_safeguard(dtype, caplog):
     # Check caplog for warning.
     msg = f"setting {num_eps} extremal filter value(s) to zero."
     assert msg in caplog.text
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_array_filter_dtype_passthrough(dtype):
+    """
+    We upcast to use scipy's fast interpolator. We do not recast
+    on exit, so this is an expected fail for singles.
+    """
+    if dtype == np.float32:
+        pytest.xfail(reason="ArrayFilter currently upcasts singles.")
```

Comment:
Add `strict=True` so that we are forced to remove this in the future.

Reply:
Sure. Done. Thanks!